DataStructure/Graph/graph.java

```diff
@@ -0,0 +1,240 @@
+package Graph;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+import Queue.Queue;
+
+public class Graph {
+    int numberofnodes;
+    private boolean vis[];
+    private boolean directed;
+    private int arr[][];
+    int numberofedges;
+
+    Graph(int noofnodes, boolean dir) {
+        this.numberofnodes = noofnodes;
+        this.directed = dir;
+        arr = new int[numberofnodes][numberofnodes];
+        vis = new boolean[numberofnodes];
+        numberofedges = 0;
+
+    }
+
```

Comment:
Add comments in the code. Like what this function does and how. Do this throughout the modules. Wherever comments are required you need to add.